services/core/shared/core/compat/sdk_v5/blockchainIndex.js

```diff
@@ -320,18 +336,27 @@ const performGenesisAccountsIndexing = async () => {
 };
 
 const indexGenesisAccounts = async () => {
-	const isScheduled = await genesisAccountsCache.get('isGenesisAccountIndexingScheduled');
+	const isGenesisAccountsAlreadyIndexed = async () => {
+		const keyValueDB = await getKeyValueStoreIndex();
+		const [{ value: isAlreadyIndexed } = {}] = await keyValueDB.find(
+			{ key: isGenesisAccountIndexingFinished, limit: 1 },
+			['value'],
+		);
+		return isAlreadyIndexed === String(true); // since 'value' is of type 'string'
+	};
+
+	const isScheduled = await genesisAccountsCache.get(isGenesisAccountIndexingScheduled);
 
-	if (isScheduled === true) {
+	if (isScheduled === true || await isGenesisAccountsAlreadyIndexed()) {
 		logger.info('Skipping genesis account index update (one-time operation, already indexed)');
 		return;
 	}
 
 	try {
 		logger.info('Attempting to update genesis account index (one-time operation)');
 		await performGenesisAccountsIndexing();
-		await genesisAccountsCache.set('isGenesisAccountIndexingScheduled', true);
-		await genesisAccountsCache.set('genesisAccountPageCached', 0);
+		await genesisAccountsCache.set(isGenesisAccountIndexingScheduled, true);
+		await genesisAccountsCache.set(genesisAccountPageCached, 0);
```

Comment:
Don't we need to completely move account index information from redis to MySQL)?

Reply:
@MichalTuleja Please confirm.

Reply:
I think that would even work assuming that information reflects the number of imported accounts, not the number of scheduled accounts. That's why I see an issue here: the import is scheduled but not yet in the DB snapshot. For the DB snapshot we need another flag that indicates the account import has already finished. Is the `genesisAccountPageCached` indicating the number of page cached or scheduled?

Reply:
The Redis dependency has now been completely removed from the process.